### PR TITLE
Update GTFS-lib from v1.1.0 to v5.0.7

### PR DESCRIPTION
**Pull request by floscher**
I noticed that you use an old version of [GTFS-lib](https://github.com/conveyal/gtfs-lib/releases).

<del>This PR updates the dependency from `v1.1.0` to `v3.3.0`. But note that it does not deal with compilation issues that this causes, because the API of that library changed.</del>

<del>I'm not really familiar with this library or GTFS processing in general, so it's probably better if I leave the task of adapting the plugin to that newer version up to you :wink:.</del>

### CODE REVIEW

File: src/main/java/org/matsim/contrib/gtfs/LineBrowser.java [hunk 1]
```diff
@@ -119,40 +122,41 @@ private void tripStage(GTFSFeed feed, MyPattern pattern, Stage stage) {
...
-				routeTreeItems.put(route.route_id, routeTreeItem);
+				final MyRoute myRoute = new MyRoute(route);
+				final TreeItem<RootPattern> routeTreeItem = new TreeItem<>(myRoute);
+				routeTreeItems.put(route.route_id, new Pair<>(routeTreeItem, myRoute));
 				agencyTreeItems.get(route.agency_id).getChildren().add(routeTreeItem);
 			}
 			for (Pattern pattern : feed.patterns.values()) {
-				for (String routeId : pattern.associatedRoutes) {
-					((MyRoute) routeTreeItems.get(routeId).getValue()).getPatterns().add(pattern);
+				for (String routeId : pattern.associatedTrips.stream().map(it -> feed.trips.get(it).route_id).collect(Collectors.toSet())) {
```
Comment (floscher) [inline src/main/java/org/matsim/contrib/gtfs/LineBrowser.java]: This is the main line that needed changing from v1.1.0 to v5.0.7 of `gtfs-lib`. I looked at
https://github.com/conveyal/gtfs-lib/blob/1438918bda9506727317c8328287d1410b1b7071/src/main/java/com/conveyal/gtfs/model/Pattern.java#L43-L46
and replaced `associatedRoutes` with something similar.

The other changed lines in this file are introducing generics to the `TreeItem`s.

**Comment by floscher** [comment]
Turns out updating the `gtfs-lib` didn't break the plugin much. It was essentially one line in `LineBrowser` that needed a change. So the current state of this PR should be ready to merge.

I restored the dependency on the JOSM plugin `geotools`. The issue was that starting with JTS version `1.15.0`, the package names changed, so different dependencies and also this plugin mixed classes from both packages, which led to conflicts. But now everything should be fine with the JOSM plugin `geotools` also using the newer version of JTS. See also conveyal/gtfs-lib#97 for similar problems that occur(ed) in that project.

**Comment by nkuehnel** [comment]
Thank you so much, highly appreciated!

**Comment by floscher** [comment]
@nkuehnel You're welcome.

By the way, if you create a new release, make sure you use [annotated tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging#_annotated_tags) by adding the option `-a` or `-s` to the `git tag`. This makes sure the version numbers are constructed correctly.

**Pull request merged**